lib/eco_service_web/live/eco_service_live/community_list.ex

```diff
@@ -4,45 +4,73 @@ defmodule EcoServiceWeb.EcoServiceLive.CommunityList do
   alias EcoService.EcoServiceContext
 
   def mount(_params, _session, socket) do
-    params = %{limit: 10, offset: 0}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params = %{limit: 10, offset: 0}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
 
     {:ok,
     socket
     |> assign(:communities, all_communities)
-    |> assign(:limit, params.limit)
-    |> assign(:offset, params.offset)
-    |> assign(:all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
+    |> assign(:limit, pag_params.limit)
+    |> assign(:offset, pag_params.offset)
+    |> assign(:count_of_all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
     }
   end
 
   def handle_event("prev", _params, socket) do
-    params= %{limit: 10,  offset: socket.assigns.offset-10}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params= %{limit: 10,  offset: socket.assigns.offset-10}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
 
     {:noreply,
     socket
     |> assign(:communities, all_communities)
-    |> assign(:limit, params.limit)
-    |> assign(:offset, params.offset)
-    |> assign(:all_records, nil)
-    |> assign(:all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
+    |> assign(:limit, pag_params.limit)
+    |> assign(:offset, pag_params.offset)
+    |> assign(:count_of_all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
    }
   end
 
   def handle_event("next", _params, socket) do
-    params= %{limit: 10,  offset: socket.assigns.offset+10}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params= %{limit: 10,  offset: socket.assigns.offset+10}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
 
     {:noreply,
     socket
     |> assign(:communities, all_communities)
-    |> assign(:limit, params.limit)
-    |> assign(:offset, params.offset)
-    |> assign(:all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
+    |> assign(:limit, pag_params.limit)
+    |> assign(:offset, pag_params.offset)
+    |> assign(:count_of_all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
    }
   end
 
+  def handle_event("first", _params, socket) do
+    pag_params = %{limit: 10, offset: 0}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
+
+    {:noreply,
+    socket
+    |> assign(:communities, all_communities)
+    |> assign(:offset, pag_params.offset)
+   }
+  end
+
+  def handle_event("last", _params, socket) do
+    all_communities_count = Enum.count(EcoServiceContext.fetch_all_communities())
+
+    offset = all_communities_count - socket.assigns.limit
```

Comment:
Good logical thinking skills shown here!!

Reply:
Thank you